### PR TITLE
postfix dereference syntax

### PR DESCRIPTION
**Pull request by briandfoy**
Recognize the postfix dereference syntax

### CODE REVIEW

**Comment by moznion** [comment]
Wow, it's really great work!
Looks good.

File: gen/gen_decl.pl [hunk 1]
```diff
@@ -639,3 +639,11 @@ package Compiler::Lexer::Kind;
 Verbose             Comment
 Verbose             WhiteSpace
 Undefined       	Undefined           	
+Symbol              PostDeref
```
Comment (goccy) [inline gen/gen_decl.pl]: Please put new definition above 'Undefined'. 'Undefined' is terminal symbol.

File: src/compiler/lexer/Compiler_lexer.cpp [hunk 2]
```diff
@@ -46,6 +47,7 @@ Tokens *Lexer::tokenize(char *script)
 	char ch = smgr->currentChar();
 	for (; ch != EOL; smgr->idx++) {
 		ch = smgr->currentChar();
+		// cout << "Char is " << ch << endl;
```
Comment (goccy) [inline src/compiler/lexer/Compiler_lexer.cpp]: Please remove comment out.

File: src/compiler/lexer/Compiler_lexer.cpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #include <lexer.hpp>
+#include <iostream>
```
Comment (goccy) [inline src/compiler/lexer/Compiler_lexer.cpp]: Please remove  this definition. Already included in 'lexer.hpp'

File: src/compiler/lexer/Compiler_scanner.cpp [hunk 4]
```diff
@@ -6,7 +6,8 @@ namespace TokenKind = Enum::Token::Kind;
 
 Scanner::Scanner() :
 	isStringStarted(false), isRegexStarted(false), isPrototypeStarted(false), isFormatStarted(false),
-	isFormatDeclared(false), commentFlag(false), hereDocumentFlag(false), skipFlag(false),
+	isFormatDeclared(false), commentFlag(false), hereDocumentFlag(false),
```
Comment (goccy) [inline src/compiler/lexer/Compiler_scanner.cpp]: Please modify to original alignment.

File: t/perl/new_syntax/perl5.10.t [hunk 5]
```diff
@@ -0,0 +1,31 @@
+use strict;
+use warnings;
+use Test::More 0.95;
+
+use_ok('Compiler::Lexer');
+
```
Comment (goccy) [inline t/perl/new_syntax/perl5.10.t]: What is this test ?

Comment (moznion) [inline t/perl/new_syntax/perl5.10.t]: Maybe these test cases are place holder, not implemented yet, I think.

**Comment by goccy** [comment]
What is 'experiments' directory ?
Also, if you can benchmark, please put it's results to this page.
I got the following results on my environment. It looks good to me. :+1: 

`$ perl example/benchmark.pl /path/to/perl-5.16.0/t/op/lex.t`

## BEFORE
```
Benchmark: timing 1000 iterations of COMPILER_LEXER, PPI...
COMPILER_LEXER:  0 wallclock secs ( 0.22 usr +  0.03 sys =  0.25 CPU) @ 4000.00/s (n=1000)
            (warning: too few iterations for a reliable count)
       PPI:  2 wallclock secs ( 1.96 usr +  0.04 sys =  2.00 CPU) @ 500.00/s (n=1000)
                 Rate            PPI COMPILER_LEXER
PPI             500/s             --           -87%
COMPILER_LEXER 4000/s           700%             --
```

## AFTER
```
Benchmark: timing 1000 iterations of COMPILER_LEXER, PPI...
COMPILER_LEXER:  0 wallclock secs ( 0.22 usr +  0.03 sys =  0.25 CPU) @ 4000.00/s (n=1000)
            (warning: too few iterations for a reliable count)
       PPI:  2 wallclock secs ( 1.95 usr +  0.04 sys =  1.99 CPU) @ 502.51/s (n=1000)
                 Rate            PPI COMPILER_LEXER
PPI             503/s             --           -87%
COMPILER_LEXER 4000/s           696%             --
```

**Comment by briandfoy** [comment]
I've fixed up those code issues.

I want to fill out the new features tests and I've added notes to each about the things to test.

**Comment by goccy** [comment]
:+1: 
Thanks, I expect to your support of new syntax !

**Pull request merged**